components/sync_manager/src/manager.rs

```diff
@@ -32,180 +86,75 @@ impl SyncManager {
         Self::default()
     }
 
-    pub fn autofill_engine(engine: &str) -> Option<Box<dyn SyncEngine>> {
-        autofill::get_registered_sync_engine(engine)
+    pub fn autofill_engine(engine_name: &str) -> Option<Box<dyn SyncEngine>> {
+        autofill::get_registered_sync_engine(engine_name)
     }
 
-    pub fn logins_engine(engine: &str) -> Option<Box<dyn SyncEngine>> {
-        logins::get_registered_sync_engine(engine)
+    pub fn logins_engine(engine_name: &str) -> Option<Box<dyn SyncEngine>> {
+        logins::get_registered_sync_engine(engine_name)
     }
 
-    pub fn places_engine(engine: &str) -> Option<Box<dyn SyncEngine>> {
-        places::get_registered_sync_engine(engine)
+    pub fn places_engine(engine_name: &str) -> Option<Box<dyn SyncEngine>> {
+        places::get_registered_sync_engine(engine_name)
     }
 
-    pub fn tabs_engine(engine: &str) -> Option<Box<dyn SyncEngine>> {
-        tabs::get_registered_sync_engine(engine)
+    pub fn tabs_engine(engine_name: &str) -> Option<Box<dyn SyncEngine>> {
+        tabs::get_registered_sync_engine(engine_name)
     }
 
-    fn get_engine(engine: &str) -> Option<Box<dyn SyncEngine>> {
-        match engine {
-            "history" => Self::places_engine("history"),
-            "bookmarks" => Self::places_engine("bookmarks"),
-            "addresses" => Self::autofill_engine("addresses"),
-            "creditcards" => Self::autofill_engine("creditcards"),
-            "logins" => Self::logins_engine("logins"),
-            "tabs" => Self::tabs_engine("tabs"),
-            _ => {
-                log::warn!("get_engine() unknown engine: {}", engine);
-                None
-            }
+    fn get_engine(engine_id: &EngineId) -> Option<Box<dyn SyncEngine>> {
+        match engine_id {
+            EngineId::History => Self::places_engine("history"),
+            EngineId::Bookmarks => Self::places_engine("bookmarks"),
+            EngineId::Addresses => Self::autofill_engine("addresses"),
+            EngineId::CreditCards => Self::autofill_engine("creditcards"),
+            EngineId::Logins => Self::logins_engine("logins"),
```

Comment:
I decided to push the `EngineId` type into tho the `get_registered_sync_engine()` function.  This seems like the best way to avoid issues with the strings not lining up.